### ISSUE TITLE
'Post to Reddit' - Design

### ISSUE THREAD

**Issue by voythas** [Tunous/Dank#6]
Hey,
I'm sharing my first try at post submission for Dank. I have no idea how to code - I can design only.

Here's link: https://invis.io/H6PVTGBKXWT

It's a first shot at this, I tried to stick as close to original designs as I only could. I do have one or two different ideas for it, so I might upload them later.

**Comment by saket** [comment]
This looks quite nice to start with!

If it (still) matters, I was thinking that the post screen need not be a full-screen screen. It might look good as a dialog, somewhat similar to [Things](https://culturedcode.com/things/):

![392x696bb](https://user-images.githubusercontent.com/2387680/50902577-4552e100-1441-11e9-8e5a-4b8b2fb2e4c1.jpg)

People aren't going to type a lot on phone, so a dialog should have enough space for them to share a link or type a short self-text. In case users want more space, they could always switch to full-screen compose.

Another advantage is that when a link is shared to Dank from another app, the background app will still be visible behind the dialog. This gives you a feeling that you aren't leaving your current app. I'm not sure if I'm able to explain this properly, but the cognitive overhead of switching apps is higher than smaller share dialogs.

**Comment by voythas** [comment]
I fully understand what you mean. I could do this the way you shown, yeah. I will think more about that.

**Comment by Tunous** [comment]
I think that having post screen as a dialog is a good idea. I like how it looks in your designs @voythas (version without tabs).

One thing I would change here is to remove that formatting bar from the dialog for text post and instead add a button to full screen editor somewhere. So we would have 4 screens in total: dialogs for text, image and link posts plus a full-screen version for text post with all the formatting options available.

**Comment by saket** [comment]
Looks nice!

> One thing I would change here is to remove that formatting bar from the dialog for text post and instead add a button to full screen editor somewhere.

+1. Will keep this screen consistent with comment reply and message reply designs. 

The version without tabs looks cleaner to me as well, but I'm not sure how switching between link, self-text and image will look like in this case.

**Comment by voythas** [comment]
Yeah, I will put more thought to it. Expect new designs this weekend probably.

**Comment by dotvhs** [comment]
Here's new design that covers your ideas and should also resolve issue of switching between different post types: https://invis.io/H6PVTGBKXWT#/340605335_Create_Post-2-Simple_V2

**Comment by Tunous** [comment]
One thing that is missing is an option for selecting subreddit which would be needed when posting via share action or from frontpage.

**Comment by dotvhs** [comment]
> One thing that is missing is an option for selecting subreddit which would be needed when posting via share action or from frontpage.

I thought it perhaps could be a two step process in that case to not overclutter that UI? Not sure how I feel about that just yet.

**Comment by saket** [comment]
A team member from my company's design team has agreed to help me with this. He has some ideas on how the dialog could be animated as well. I'll sit with him and sketch out some ideas based on your designs, @dotvhs 

Opening a PR in the OG repository to track this: [saket/dank#7](https://github.com/saket/Dank/pull/7)

**Comment by saket** [comment]
@dotvhs not at all, please keep going. Your ideas are good and I'll probably end up using some of them. 

To give some feedback on this interaction, this is similar to what the official app does: show options for post kind (self-text, image, link) before opening the post screen. I think one problem with this is if you choose the wrong kind or change your mind afterwards, you can't change the kind without closing the post screen and clicking on FAB again. 

By the way, do you have ideas on where to fit subreddit selection here?

**Comment by dotvhs** [comment]
> To give some feedback on this interaction, this is similar to what the official app does: show options for post kind (self-text, image, link) before opening the post screen. I think one problem with this is if you choose the wrong kind or change your mind afterwards, you can't change the kind without closing the post screen and clicking on FAB again.

Yeah, I'm aware of that. The previous version had this solved by keeping current type as a button.


> By the way, do you have ideas on where to fit subreddit selection here?

I had two, but I am not very proud of them, really. One is a bit unconventional, like this this:
![fab](https://user-images.githubusercontent.com/681416/51421876-ebbb8680-1ba5-11e9-8c03-55abee348849.png)

And another is more of a workaround of the problem, which would be not displaying FAB on frontpage/popular/all/multireddits. :(
